### DISCUSSION TITLE
We are planning to replace `Rasterize` and all `Export`

### DISCUSSION THREAD

**Discussion by JerryI** [WLJSTeam/wljs-notebook#275] [Graphics & Plot functions]
It was reported multiple times, that `Export` and `Rasterize` is quite buggy in WLJS Notebook.
Our recreations of the native `Plot`, `Graphics` and `Graphics3D` are still far from being perfect. Some styling options are missing. People have been using `Export` and `Rasterize`, which produce "original" Mathematica's graphics  considering all styles.  However, now we (@KirillBelovTest  and I @JerryI ) are struggling with several problems:

1.  `Rasterize` (as well as `Export[out_, g_Graphics]`) for some reason uses `StandardForm` of expressions, which we had to alter in order to support Mathematica's syntax sugar, expressions styling and many more things in our environment. 
2. The output of `Rasterize` or `Export` does not match with a look of graphics in a notebook cell (since it uses different implementations of primitives).
3. Our implementation of graphics (what you see in a cell usually) still lacks some of the styling features
4. Fixing `Export` breaks our styling and vise versa.

From this on, we believe the way of still trying to support Mathematica's output is probably doomed. There is a suggestion, which was already successfully tested to completely clear the symbol `Rasterize` and replace it with our own, i.e.

```
ClearAll[Rasterize];
Rasterize[g_Graphics, opts__] := ...
Rasterize[g_Graphics3D, opts__] := ...
...
```

it will affect `Export` automatically. 

What it means in a nutshell:
- we rasterize the data directly from what is rendered in a cell
- it will look the same in the exported file
- we completely loose all styling options, which are still not implemented in WLJS, but were supported in Mathematica
  - PlotLegends
  - Theme
  - ...etc

Test export

<img width="442" alt="Screenshot 2024-09-02 at 22 02 35" src="https://github.com/user-attachments/assets/f6570f5d-de7a-475b-8565-4a67b9ec8d0c">

<br/>

Please share your opinion. 
I personally believe this is more progressive way, focusing on WLJS versions of `Plot`, `Graphics` and etc, improving it, adding requested style options and other features.

It would be also helpful if you could mention the styling options and any other features related to `Graphics` and `Graphics3D` which you need in your daily work. Then we can prioritize them in a right way

**Comment by JerryI** [comment]
Update 12.Sep.2024

`Rasterize` was replaced. However, we faced some limitations.
You have to redownload an app to get this options to work.

**Reply by Kri-Ol** [reply]
2.5.6 didn't work for me. No graphs

**Reply by JerryI** [reply]
> 2.5.6 didn't work for me. No graphs

Did you get any output? A source code could also be helpful

**Comment by Kri-Ol** [comment]
Was dumping what looks like internal commands/data instead of rasterized image. Uninstalled, removed all installation directory, installed again - rasterization works now. Please disregard
